edb/edgeql/compiler/stmtctx.py

```diff
@@ -139,48 +139,52 @@ def fini_expression(
     ):
         ir = setgen.scoped_set(ir, ctx=ctx)
 
+    # Collect all of the expressions stored in various side sets
+    # that can make it into the output, so that we can make sure
+    # to catch them all in our fixups and analyses.
+    extra_exprs = []
+    extra_exprs += [
+        rw for rw in ctx.env.type_rewrites.values()
+        if isinstance(rw, irast.Set)
+    ]
+    extra_exprs += [
+        p.sub_params.decoder_ir for p in ctx.env.query_parameters.values()
+        if p.sub_params and p.sub_params.decoder_ir
+    ]
+
+    all_exprs = [ir] + extra_exprs
```

Comment:
We can't get rid of `extra_exprs`, since we need to use that. We could build `extra_exprs` in one go, but I kind of like having each case more clearly specified, since it to me emphasizes that more things will be added